ipaserver/install/service.py

```diff
@@ -377,10 +377,40 @@ def add_cert_to_service(self):
             logger.critical("Could not add certificate to service %s entry: "
                             "%s", self.principal, str(e))
 
-    def import_ca_certs(self, db, ca_is_configured, conn=None):
+    def import_ca_certs_file(self, cafile, ca_is_configured, conn=None):
+        """
+        Import the CA certificates stored in LDAP into a file
+
+        :param cafile: the file to write the CA certificates to
+        :param ca_is_configured: whether IPA is CA-less or not
+        :param conn: an optional LDAP connection to use
+        """
+        if conn is None:
+            conn = api.Backend.ldap2
+
+        ca_certs = None
+        try:
+            ca_certs = certstore.get_ca_certs(
+                conn, self.suffix, self.realm, ca_is_configured)
+        except errors.NotFound:
+            pass
+        else:
+            with open(cafile, 'wb') as fd:
+                for cert, nickname, trusted, ext_key_usage in ca_certs:
+                    fd.write(cert.public_bytes(x509.Encoding.PEM))
+
+    def import_ca_certs_nssdb(self, db, ca_is_configured, conn=None):
+        """
+        Import the CA certificates stored in LDAP into an NSS database
+
+        :param db: the target NSS database
+        :param ca_is_configured: whether IPA is CA-less or not
+        :param conn: an optional LDAP connection to use
+        """
         if conn is None:
             conn = api.Backend.ldap2
 
+        ca_certs = None
```

Comment:
`ca_certs` will always be defined in the `else` branch.  Even if the universe goes wrong and it's not , this fix wouldn't make it better because `None` is not an iterable, so the `for` below would still throw an exception.

Reply:
Removed